File: nodejs/expiration.js
// Require Couchbase Module
var couchbase = require('couchbase');

// Setup Cluster Connection Object
var cluster = new couchbase.Cluster('couchbase://127.0.0.1');

// Setup Bucket object to be reused within the code
var bucket = cluster.openBucket('travel-sample');

// Setup a new Document, with 2 second expiry and store in in the Bucket
//		 - note -  API uses unix epoch time in seconds
var key = "nodeDevguideExampleExpiry";
bucket.insert(key, {test:"Some Test Value"}, {expiry: 2}, function(err, res) {
    if (err) throw err;

    if(res){

        console.log('Initialized Document With Expiry');

        // Get Document before Expiry
        bucket.get(key, function(err, resPre) {
            if (err) throw err;

            if(resPre) {

                console.log('Retreived Document Before Expiry:', resPre.value);

                // Wait 4 seconds, and try to retrieve Document again
                setTimeout(function () {
                    bucket.get(key, function (err, resExp) {
                        if (err){

                            // Document expired, and should throw an error of not found
                            console.log('Document Expired:', err.message)
                            console.log('Example Successful - Exiting');
                            process.exit(0);
                        }
                    });
                }, 4000);
            }
        });
    }
});

File: go/expiration.go
package main

import (
	"fmt"
	"time"

	"gopkg.in/couchbase/gocb.v1"
)

// bucket reference - reuse as bucket reference in the application
var bucket *gocb.Bucket

func main() {
	// Connect to Cluster
	cluster, err := gocb.Connect("couchbase://127.0.0.1")
	if err != nil {
		fmt.Println("ERROR CONNECTING TO CLUSTER:", err)
	}
	// Open Bucket
	bucket, err = cluster.OpenBucket("travel-sample", "")
	if err != nil {
		fmt.Println("ERROR OPENING BUCKET:", err)
	}

	// Create a document and assign an expiration of 2 seconds
	key := "goDevguideExampleExpiration"
	val := "Expiration Test Value"
	_, err = bucket.Upsert(key, &val, 2)
	if err != nil {
		fmt.Println("ERROR CREATING DOCUMENT:", err)
	}

	// Retrieve Value Right Away
	var retValue interface{}
	_, err = bucket.Get(key, &retValue)
	if err != nil {
		fmt.Println("ERROR RETURNING DOCUMENT:", err)
	}
	fmt.Println("Document Not Yet Expired:", retValue)

	// Sleep for 4 seconds
	time.Sleep(4 * time.Second)

	// Try to retrieve document when Expired
	_, err = bucket.Get(key, &retValue)
	if err != nil {
		fmt.Println("Document Expired:", err)
	}

	// Create a document with no expiration, and add an expiry using
	//  touch after the document is created of 2 seconds
	_, err = bucket.Upsert(key, &val, 2)
	if err != nil {
		fmt.Println("ERROR CREATING DOCUMENT:", err)
	}

	// Add an expiry
	_, err = bucket.Touch(key, 0, 2)
	if err != nil {
		fmt.Println("ERROR TOUCHING DOCUMENT:", err)
	}

	// Retrieve Document
	_, err = bucket.Get(key, &retValue)
	if err != nil {
		fmt.Println("ERROR RETURNING DOCUMENT:", err)
	}
	fmt.Println("Document Not Yet Expired:", retValue)

	// Sleep for 4 seconds
	time.Sleep(4 * time.Second)

	// Try to retrieve document when Expired
	_, err = bucket.Get(key, &retValue)
	if err != nil {
		fmt.Println("Document Expired:", err)
	}

	// Exiting
	fmt.Println("Example Successful - Exiting")
}
